apps/api/src/domain/models/song/commands/add-song.command-handler.ts

```diff
@@ -0,0 +1,48 @@
+import { Ack, CommandHandler, ICommandHandler } from '@coscrad/commands';
+import { buildSimpleValidator } from '@coscrad/validation';
+import { InternalError, isInternalError } from '../../../../lib/errors/InternalError';
+import { NotFound } from '../../../../lib/types/not-found';
+import { RepositoryProvider } from '../../../../persistence/repositories/repository.provider';
+import getInstanceFactoryForEntity from '../../../factories/getInstanceFactoryForEntity';
+import { ResourceType } from '../../../types/ResourceType';
+import { Song } from '../song.entity';
+import { AddSong } from './add-song.command';
+
+@CommandHandler(AddSong)
+export class AddSongHandler implements ICommandHandler {
+    constructor(private readonly repositoryProvider: RepositoryProvider) {}
+
+    async execute(command: AddSong): Promise<Ack | Error> {
+        // Validate type
+        const typeErrors = buildSimpleValidator(AddSong)(command).map(
```

Comment:
This is common in functional programming. `buildSimpleValidator(AddSong)` takes in the constructor `AddSong` and builds from it a `simple validator` (which I should call `Simple Validation Function`. That itself takes in an unknown and returns an array of errors which is populated in case the command instance has any type errors.

I could've stored an intermediate reference
```ts
const validateCommandType = buildSimpleValidator(AddSong);

const typeErrors = validateCommandType(command).map(...)
```

but that reference does absolutely nothing except to expand the surface area of (mutable) constants within the current scope.